### ISSUE TITLE
from (0.50.0) to (0.51.0) bundle-size is increased

### ISSUE THREAD

**Issue by adnan1naeem** [rollup/rollup#1705]
I have updated my version from 0.50.0 to 0.51.0.when i ran the build my bundle size of package is increased.kindly tell me what are new updated in our new  package so we can add them 
My bundle size in version 0.50.0 is  17.78 kb:
![screenshot from 2017-11-08 17-49-19](https://user-images.githubusercontent.com/22443038/32550044-73003d80-c4ad-11e7-8024-64e33cf36a1a.png)

My bundle size after upgrading Package to 0.51.0 is 25..11 kb
![screenshot from 2017-11-08 17-48-21](https://user-images.githubusercontent.com/22443038/32550087-a5e7a5e4-c4ad-11e7-97b5-5807e54db4be.png)

**Comment by lukastaegert** [comment]
Hi,
it is difficult to tell why the bundle size increased but the most probable reason is that new logic for getters. Whenever rollup encounters an access to a property of an object is does not know, it assumes this property might be a getter and accessing it may have side-effects.

In some cases (e.g. lots of external or global variables), this can lead to many more code that is now retained. Since a correct bundle is our most important goal, we will keep this logic as the default. However I understand that most projects do not use getters so we might add a new configuration option that disables this logic.

I cannot promise a definite time frame but I will look into preparing a pull request to add such a flag during the next week. Until then if bundle size is very important to you, please stick with rollup@0.50.1.

**Comment by lukastaegert** [comment]
Hi @adnan1naeem, could you please check how your bundle size changes if you install rollup via
```
npm install rollup/rollup#more-treeshaking-options
```
and add
```
{
...
  treeshake: {
    propertyReadSideEffects: false
  }
...
}
```
to your options? If you use the CLI interface, you could also specify `--no-treeshake.propertyReadSideEffects`.

**Comment by lukastaegert** [comment]
This has now been officially released as Rollup v0.52. I would like to know if with the new `treeshake.propertyReadSideEffects` option there are still people who receive larger bundles than with Rollup v0.50 or earlier. Otherwise I will close this issue soon.

**Comment by lukastaegert** [comment]
As there has been no further communication, I will close this issue for now.
